src/netbase.cpp

```diff
@@ -68,6 +68,8 @@ bool static LookupIntern(const char *pszName, std::vector<CNetAddr>& vIP, unsign
 
     {
         CNetAddr addr;
+        // We can't actually resolve onion addresses, return a network address
```

Comment:
I think it would be clearer to point out that we treat ".onion" addresses the same way as we treat IPv4 a.b.c.d or IPv6 hex addresses, with the exception that the standard library resolving function don't support them.

Reply:
> I think it would be clearer to point out that we treat ".onion" addresses the same way as we treat IPv4 a.b.c.d or IPv6 hex addresses

Not 100% sure what you mean here, specifically, what is your specific meaning of "treat the same way"?

Reply:
For example "1.2.3.4" is a name, but it's one that directly encodes an IP address; no resolving is done when you turn it into a CNetAddr, it's just converted. The same is true for onion addresses; from our perspective, those are not hostnames - they're direct encodings of CNetAddr for onion addresses.

In contrast, "sipa.be" is a hostname that needs network activity to resolve to a CNetAddr.

Reply:
Updated: https://github.com/bitcoin/bitcoin/pull/15824/files#diff-4942713bcd2752bb531bf76c9a0ba38eR71